### PR TITLE
Put Origin in GA

### PR DESCRIPTION
**Pull request by tlwr**
This PR is branched off #337 - review that first

What
----

Put the authenticated user's `origin` from UAA in a Google analytics dimension, if it exists.

How to review
-------------

Code review

[Read](https://developers.google.com/analytics/devguides/collection/gtagjs/custom-dims-mets) on how `gtag` deals with custom dimensions

See Concourse

Who can review
---------------

@richardTowers

### CODE REVIEW

**Comment by tlwr** [comment]
Rebased so the diff is more obvious

File: src/components/app/app.test.ts [hunk 1]
```diff
@@ -208,6 +209,14 @@ describe('app test suite', () => {
 
       expect(response.redirect).toEqual(orgs.definition.path);
     });
+
+    it('should add a meta tag for origin', async () => {
+      const orgs = router.findByName('admin.organizations');
+      const response = await agent.get(orgs.definition.path);
+
+      expect(response.status).toEqual(200);
+      expect(response.text).toMatch('<meta name="origin" content="uaa" />');
```
Comment (richardTowers) [inline src/components/app/app.test.ts]: I think `origin` might not be sufficiently specific for a meta tag (e.g. one might confuse it with "origin" as in "same origin policy") maybe `x-user-identity-origin` or something?

Comment (tlwr) [reply src/components/app/app.test.ts]: Agreed, resolved in 75d7c63f323ae29cae6759f09afbd1004edf659d

File: src/layouts/govuk.njk [hunk 2]
```diff
@@ -14,17 +14,26 @@
...
-  </script>
+    var dimensions = {};
 
-  <meta name="csrf-token" content="{{context.csrf.toString()}}" />
+    var originTag = document.querySelector('meta[name="x-user-identity-origin"]')
+    if (originTag && originTag.content) {
+      dimensions['dimension1'] = originTag.content;
+    }
+
+    gtag('config', 'UA-43115970-5', dimensions);
```
Comment (richardTowers) [inline src/layouts/govuk.njk]: I'm not sure this is right according to the google docs [you linked](https://developers.google.com/analytics/devguides/collection/gtagjs/custom-dims-mets):

```js
// Configures custom dimension<Index> to use the custom parameter
// 'dimension_name' for 'GA_MEASUREMENT_ID', where <Index> is a number
// representing the index of the custom dimension.
gtag('config', 'GA_MEASUREMENT_ID', {
  'custom_map': {'dimension<Index>': 'dimension_name'}
});

// Sends the custom dimension to Google Analytics.
gtag('event', 'any_event_name', {'dimension_name': dimension_value});
```

Shouldn't it be:

```js
gtag('config', 'UA-43115970-5', {
  'custom_map': {'dimension1': 'origin'}
});
gtag('event', 'origin_dimension', {'origin': originTag.content});
```

Or something? I'm not 100% sure - the docs are quite confusing.

Comment (tlwr) [reply src/layouts/govuk.njk]: After reviewing the page I linked I realised I linked a not-helpful page.

`custom_map` is renaming things in the console, along with the implicit event.

My understanding is calling `gtag` with `'config'` as the first argument does two things:

- configure `gtag` with the ID (`UA-*`) and with any additional properties
- implicitly send a pageview event

So calling `'config'` and then calling `'event'` AFAICT is two events, but we only want one.

I think what we want is:

```
gtag('set', 'dimension1', originTag.content);
gtag('config', 'UA-43115970-5')
```

which is longhand for

```
gtag('config', 'UA-43115970-5', dimensions);
```

---

Info about `'set'`, `'event'`, and `'config'` [here](https://developers.google.com/gtagjs/reference/api)

Gohugo.io uses Google analytics in this fashion, search for `dimension1` within `view-source:https://gohugo.io/`

**Pull request merged**